scripts/main.js

```diff
@@ -308,9 +314,89 @@ window.addEventListener('WebChannelMessageToContent', function (event) {
       selectNextItem(-1);
     }
   }
+
+  // collecting session metrics on open and closed message types
+  if (message.type === 'popupopen') {
+
+    sessionMetrics.set({startTime: new Date().getTime()});
+
+  } else if (message.type === 'popupclose') {
+
+    sessionMetrics.set({endTime: new Date().getTime()}); 
+    sessionMetrics.set({duration:sessionMetrics.getDuration()});
+
+    if (sessionMetrics.attributes.hasEngaged) {
+
+      var selectedGroup = $('li.selected').parent().parent();
+      var resultIndex = selectedGroup.find('.selected').index();
+
+      sessionMetrics.set({resultType: selectedGroup.attr('data-region')});
+      sessionMetrics.set({resultIndex: resultIndex});
+
+    }
+
+    console.log(sessionMetrics.attributes);
```

Comment:
Assuming we actually want to save these metrics somewhere, for the prototype, the simplest thing is to xhr the data somewhere (maybe stand up a microservice). I don't know if the xhr will complete after the popup closes.

For production, we'll probably need to ship over a WebChannel message with the fields in it, then have gecko attach those datapoints to the telemetry service. We could use a pref for the prototype to redirect our data to a test service, to make sure stuff is working. I'm happy to work on this, too. Let's talk about it next week?